src/main/java/com/axis/jenkins/plugins/eiffel/eiffelbroadcaster/RunListenerImpl.java

```diff
@@ -51,6 +59,14 @@ public RunListenerImpl() {
     public void onStarted(Run r, TaskListener listener) {
         UUID targetEvent = EiffelJobTable.getInstance().getEventTrigger(r.getQueueId());
         EiffelActivityStartedEvent event = new EiffelActivityStartedEvent(targetEvent);
+
+        RunUris uris = getRunUris(r);
```

Comment:
Why not static helper methods, which takes the Run and returns the URL you want? Not sure why we want to create POJO like objects in this case? I like static methods when available. Helps testability.

Reply:
I was trying to avoid boring code duplication but I was perhaps overzealous. While trying to explain myself I came up with a rather obvious solution that avoids POJOs as well as duplication.